actionSHARK/utils.py

```diff
@@ -0,0 +1,148 @@
+from typing import Optional, Any, Tuple, List
+from mongoengine import ObjectIdField
+from pycoshark.mongomodels import Commit, JobStep, RunPullRequest
+import sys
+import datetime as dt
+
+
+def parse_date(value: Optional[str] = None, is_millisecond: bool = False
+) -> Optional[dt.datetime]:
+    """Convert Datetime string to actual Datetime object.
+
+    Args:
+        value (str): The datetime string to be converted. Defaults to None.
+        is_millisecond (bool, optional): If datetime has millisecond in the string. Defaults to False.
+
+    Returns:
+        datetime: datetime formated or None if no string passed.
+    """
+    if not value:
+        return None
+
+    date_fmt = "%Y-%m-%dT%H:%M:%S%z"
+    if is_millisecond:
+        date_fmt = "%Y-%m-%dT%H:%M:%S.%f%z"
+
+    if sys.version_info.minor < 8:
```

Comment:
This should be checked together with the rest of the version information, e.g., major as it only applies to a specific major.